doc/contribute/nest_with_ides.rst

```diff
@@ -0,0 +1,103 @@
+* Does not support debugging on macOS (because Eclipse does not support lldb)
+* Does not read the NEST `.clang-format` file, so code formatting may
+  be incorrect
+* I experimented with better CMake support using ``cmake4eclipse``,
+  but did not figure it out. So instructions below rely on running
+  ``cmake`` manually in the build directory.
+* Tested with Eclipse 2020-12
+
+Preparations
+~~~~~~~~~~~~
+
+#. Install Eclipse from `Eclipse Installer <https://www.eclipse.org/downloads/packages/installer>`_.
+   Using the Installer ensures that a suitable Java is installed. The instructions
+   below are based on choosing the *Eclipse IDE for Scientific Computing*.
+#. From the Eclipse Marketplace, install *PyDev* and *LiClipse Text* extensions.
+#. In Eclipse preferences, under ``General > Security > Secure Storage > Advanced``
+   you should replace the default password encryption scheme to a more secure level
+   than default, e.g. to ``PBE...SHA512...AES_256``.
+
+Setting up the project
+~~~~~~~~~~~~~~~~~~~~~~
+
+#. Clone NEST onto your computer
+#. Build NEST manually
+
+   a. Create a build directory
+   #. Run ``cmake`` with suitable options
+   #. Run ``make all`` and ``make install``
+#. In Eclipse
+
+   a. choose ``New > Makefile project with existing code``
+   #. select the directory containing the NEST source code
+   #. choose, e.g., the ``Cross GCC Toolchain``
+#. Right click the project and choose ``Properties`` from the context
+   menu
+
+   a. Under ``C/C++ Build/Build Variables`` define ``BUILD_DIR`` and ``CONDA_ENV``,
+      both of type ``Path``. The first should contain the full path to the build
+      directory you created above, the second the full path to your conda 
+      environment, usually something like ``.../miniconda3/envs/nest-dev``.
+   #. Under ``C/C++ Build > Builder Settings``
+
+      #. uncheck ``Use default build command``
+      #. set ``Build Command`` to ``make -k -j4 all install`` (adjust
+	 number of processes to your situation)
+      #. set ``Build Directory`` to ``${BUILD_DIR}``
+   #. Under ``C/C++ Build > Environment`` prepend
+      ``${CONDA_ENV}/bin`` to ``PATH``
+   #. Under ``C/C++ General > Paths and Symbols > Includes`` add the
+      following two direcories
+      * ``${BUILD_DIR}/libnestutil`` (contains ``config.h``)
+      * ``${CONDA_ENV}/include`` (all headers from packages provided in conda environment)
+   #. Under ``PyDev - Interpreter/Grammar`` choose the intepreter from
+      your Conda environment (you may need to add it by following the
+      ``Click here to configure an interpreter not listed`` link and
+      then ``Browse for python/pypy exe`` (this temporarily takes you
+      to the global Eclipse preferences in a separate window).
+   #. Under ``Run/Debug Settings``, add a ``New ...`` launch
+      configuration, entering in ``C/C++ Application`` the full path
+      to the installed ``nest`` executable.
+
+What you can do then
```

Comment:
Perhaps we could rename this to "usage"?

Reply:
```suggestion
Usage
```